pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -254,7 +254,7 @@ func calculateIPOffset(base *big.Int, ip net.IP) int {
 func RangeSize(subnet *net.IPNet) int64 {
 	ones, bits := subnet.Mask.Size()
 	if (bits - ones) >= 31 {
-		panic("masks greater than 31 bits are not supported")
+		panic("masks shorter than /2 for IPv4 or /98 for IPv6 are not supported.")
```

Comment:
Sure.  I'm just pointing out (the subtle point) that the method is returning the number of hosts available for the subnet, and wondering if the error message that would get returned, should focus on that, rather than on mask size. For example:
```
err: = fmt.Sprintf("too many hosts allowed for subnet (%v). Must be /%d or longer", subnet, bits-30)
```
However, see my last comment on L254, where maybe this should be redesigned to check for valid subnet lengths, rather than returning the number of hosts and having the caller doing some of the checking, with the rest done here. Then, the error messages could be in terms of mask size (from tests for it being too short or too long).